File: csharp/src/GibberishRewriter.Core/WordList.cs
namespace GibberishRewriter.Core;

/// <summary>Lowercase English words from shared/wordlists/english.txt.</summary>
public sealed class WordList
{
    private readonly HashSet<string> _words;

    private WordList(HashSet<string> words) => _words = words;

    public static WordList Load(string path) => FromWords(File.ReadLines(path));

    public static WordList FromWords(IEnumerable<string> words) =>
        new(new HashSet<string>(
            words.Select(word => word.Trim()).Where(word => word.Length > 0),
            StringComparer.Ordinal));

    public int Count => _words.Count;

    /// <summary>Exact, case-sensitive lookup. The list holds lowercase words, so lowercase first.</summary>
    public bool Contains(string word) => _words.Contains(word);
}

File: python/src/gibberish_rewriter/core/word_list.py
"""Lowercase English words from shared/wordlists/english.txt."""

from __future__ import annotations

from collections.abc import Iterable
from pathlib import Path


class WordList:
    def __init__(self, words: set[str]) -> None:
        self._words = words

    @classmethod
    def load(cls, path: str | Path) -> WordList:
        with open(path, encoding="utf-8-sig") as lines:
            return cls.from_words(lines)

    @classmethod
    def from_words(cls, words: Iterable[str]) -> WordList:
        return cls({word.strip() for word in words if word.strip()})

    @property
    def count(self) -> int:
        return len(self._words)

    def contains(self, word: str) -> bool:
        """Exact, case-sensitive lookup. The list holds lowercase words, so lowercase first."""
        return word in self._words
